kv/kvclient/kvcoord: TestDistSenderCircuitBreakerModes failed

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://mesolite.cluster.engflow.com/invocations/default/a6547cf2-cc18-464c-ae6d-bbbb26348509?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on master @ [4f654c7cbd768c77d4eb539f7b96c52088536009](https://github.com/cockroachdb/cockroach/commits/4f654c7cbd768c77d4eb539f7b96c52088536009):


```
        	            	  |   | 	pkg/rpc/auth.go:90
        	            	  |   | google.golang.org/grpc.getChainUnaryHandler.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
        	            	  |   | 	pkg/rpc/context.go:132
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
        	            	  |   | 	pkg/util/stop/stopper.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
        	            	  |   | 	pkg/rpc/context.go:130
        	            	  |   | google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1145
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler
        	            	  |   | 	bazel-out/k8-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10681
        	            	  |   | google.golang.org/grpc.(*Server).processUnaryRPC
        	            	  |   | 	external/org_golang_google_grpc/server.go:1335
        	            	  |   | google.golang.org/grpc.(*Server).handleStream
        	            	  |   | 	external/org_golang_google_grpc/server.go:1712
        	            	  |   | google.golang.org/grpc.(*Server).serveStreams.func1.1
        	            	  |   | 	external/org_golang_google_grpc/server.go:947
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (15.28s)
=== RUN   TestDistSenderCircuitBreakerModes/scratchRange=true
    --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true (44.55s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45111

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://tanzanite.cluster.engflow.com/invocations/default/292a8be9-004e-4f10-ae7d-1d648235cd0a?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):


```
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1739
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1258
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*CrossRangeTxnWrapperSender).Send
        	            	  |   | 	pkg/kv/db.go:224
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender
        	            	  |   | 	pkg/kv/db.go:1133
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).send
        	            	  |   | 	pkg/kv/db.go:1116
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
        	            	  |   | 	pkg/kv/db.go:965
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Run
        	            	  |   | 	pkg/kv/db.go:988
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Get
        	            	  |   | 	pkg/kv/db.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestDistSenderCircuitBreakerModes.func1.1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go:205
        	            	  |   | github.com/cockroachdb/cockroach/pkg/testutils.RunValues[...].func1
        	            	  |   | 	pkg/testutils/subtest.go:23
        	            	  |   | testing.tRunner
        	            	  |   | 	GOROOT/src/testing/testing.go:1689
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (19.39s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18013269?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18013269?buildTab=artifacts#/) on master @ [0ea9ae61bfe16ba95414c287753c67334e92e578](https://github.com/cockroachdb/cockroach/commits/0ea9ae61bfe16ba95414c287753c67334e92e578):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.RunValues[...].func1
        	            	  | 	pkg/testutils/subtest.go:23
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (2) aborted in DistSender
        	            	Wraps: (3) result is ambiguous: context deadline exceeded
        	            	Wraps: (4) secondary error attachment
        	            	  | aborted during DistSender.Send: context deadline exceeded
        	            	  | (1)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:3116
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2203
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).divideAndSendBatchToRanges
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1739
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1258
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*CrossRangeTxnWrapperSender).Send
        	            	  |   | 	pkg/kv/db.go:224
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender
        	            	  |   | 	pkg/kv/db.go:1133
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).send
        	            	  |   | 	pkg/kv/db.go:1116
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
        	            	  |   | 	pkg/kv/db.go:965
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Run
        	            	  |   | 	pkg/kv/db.go:988
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Get
        	            	  |   | 	pkg/kv/db.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestDistSenderCircuitBreakerModes.func1.1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go:205
        	            	  |   | github.com/cockroachdb/cockroach/pkg/testutils.RunValues[...].func1
        	            	  |   | 	pkg/testutils/subtest.go:23
        	            	  |   | testing.tRunner
        	            	  |   | 	GOROOT/src/testing/testing.go:1689
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (2) aborted during DistSender.Send
        	            	  | Wraps: (3) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (13.80s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://tanzanite.cluster.engflow.com/invocations/default/19c59fd8-74b2-488c-98d1-c5bf93a4d383?testReportRun=14&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):


```
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
        	            	  |   | 	pkg/rpc/auth.go:90
        	            	  |   | google.golang.org/grpc.getChainUnaryHandler.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
        	            	  |   | 	pkg/rpc/context.go:132
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
        	            	  |   | 	pkg/util/stop/stopper.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
        	            	  |   | 	pkg/rpc/context.go:130
        	            	  |   | google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1145
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler
        	            	  |   | 	bazel-out/k8-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10681
        	            	  |   | google.golang.org/grpc.(*Server).processUnaryRPC
        	            	  |   | 	external/org_golang_google_grpc/server.go:1335
        	            	  |   | google.golang.org/grpc.(*Server).handleStream
        	            	  |   | 	external/org_golang_google_grpc/server.go:1712
        	            	  |   | google.golang.org/grpc.(*Server).serveStreams.func1.1
        	            	  |   | 	external/org_golang_google_grpc/server.go:947
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (13.93s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=14</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18021031?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18021031?buildTab=artifacts#/) on master @ [b499b3a5e08d3b10d4d542ddb2f100f35838566f](https://github.com/cockroachdb/cockroach/commits/b499b3a5e08d3b10d4d542ddb2f100f35838566f):


```
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
        	            	  |   | 	pkg/rpc/auth.go:90
        	            	  |   | google.golang.org/grpc.getChainUnaryHandler.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
        	            	  |   | 	pkg/rpc/context.go:132
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
        	            	  |   | 	pkg/util/stop/stopper.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
        	            	  |   | 	pkg/rpc/context.go:130
        	            	  |   | google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1145
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler
        	            	  |   | 	bazel-out/aarch64-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10679
        	            	  |   | google.golang.org/grpc.(*Server).processUnaryRPC
        	            	  |   | 	external/org_golang_google_grpc/server.go:1335
        	            	  |   | google.golang.org/grpc.(*Server).handleStream
        	            	  |   | 	external/org_golang_google_grpc/server.go:1712
        	            	  |   | google.golang.org/grpc.(*Server).serveStreams.func1.1
        	            	  |   | 	external/org_golang_google_grpc/server.go:947
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (13.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18021670?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18021670?buildTab=artifacts#/) on master @ [079a18ec052897ad5b240ff5b0f75b495027b87f](https://github.com/cockroachdb/cockroach/commits/079a18ec052897ad5b240ff5b0f75b495027b87f):


```
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
        	            	  |   | 	pkg/rpc/auth.go:90
        	            	  |   | google.golang.org/grpc.getChainUnaryHandler.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
        	            	  |   | 	pkg/rpc/context.go:132
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
        	            	  |   | 	pkg/util/stop/stopper.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
        	            	  |   | 	pkg/rpc/context.go:130
        	            	  |   | google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1145
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler
        	            	  |   | 	bazel-out/aarch64-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10679
        	            	  |   | google.golang.org/grpc.(*Server).processUnaryRPC
        	            	  |   | 	external/org_golang_google_grpc/server.go:1335
        	            	  |   | google.golang.org/grpc.(*Server).handleStream
        	            	  |   | 	external/org_golang_google_grpc/server.go:1712
        	            	  |   | google.golang.org/grpc.(*Server).serveStreams.func1.1
        	            	  |   | 	external/org_golang_google_grpc/server.go:947
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (12.34s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18022653?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18022653?buildTab=artifacts#/) on master @ [6a61ee06779f1e6e0fe68be4286cf4cac3e8aa4e](https://github.com/cockroachdb/cockroach/commits/6a61ee06779f1e6e0fe68be4286cf4cac3e8aa4e):


```
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
        	            	  |   | 	pkg/rpc/auth.go:90
        	            	  |   | google.golang.org/grpc.getChainUnaryHandler.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
        	            	  |   | 	pkg/rpc/context.go:132
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
        	            	  |   | 	pkg/util/stop/stopper.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
        	            	  |   | 	pkg/rpc/context.go:130
        	            	  |   | google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1145
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler
        	            	  |   | 	bazel-out/aarch64-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10679
        	            	  |   | google.golang.org/grpc.(*Server).processUnaryRPC
        	            	  |   | 	external/org_golang_google_grpc/server.go:1335
        	            	  |   | google.golang.org/grpc.(*Server).handleStream
        	            	  |   | 	external/org_golang_google_grpc/server.go:1712
        	            	  |   | google.golang.org/grpc.(*Server).serveStreams.func1.1
        	            	  |   | 	external/org_golang_google_grpc/server.go:947
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (12.69s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18032655?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18032655?buildTab=artifacts#/) on master @ [c8d47d0123555cf64cfec6c55c29d0e8e63986eb](https://github.com/cockroachdb/cockroach/commits/c8d47d0123555cf64cfec6c55c29d0e8e63986eb):


```
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
        	            	  |   | 	pkg/rpc/auth.go:90
        	            	  |   | google.golang.org/grpc.getChainUnaryHandler.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
        	            	  |   | 	pkg/rpc/context.go:132
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
        	            	  |   | 	pkg/util/stop/stopper.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
        	            	  |   | 	pkg/rpc/context.go:130
        	            	  |   | google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1145
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler
        	            	  |   | 	bazel-out/aarch64-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10679
        	            	  |   | google.golang.org/grpc.(*Server).processUnaryRPC
        	            	  |   | 	external/org_golang_google_grpc/server.go:1335
        	            	  |   | google.golang.org/grpc.(*Server).handleStream
        	            	  |   | 	external/org_golang_google_grpc/server.go:1712
        	            	  |   | google.golang.org/grpc.(*Server).serveStreams.func1.1
        	            	  |   | 	external/org_golang_google_grpc/server.go:947
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (13.89s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18041024?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18041024?buildTab=artifacts#/) on master @ [a23be6bce928c3e08074a15815b3c67a657bb40e](https://github.com/cockroachdb/cockroach/commits/a23be6bce928c3e08074a15815b3c67a657bb40e):


```
        	            	  |   | 	bazel-out/aarch64-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10679
        	            	  |   | google.golang.org/grpc.(*Server).processUnaryRPC
        	            	  |   | 	external/org_golang_google_grpc/server.go:1335
        	            	  |   | google.golang.org/grpc.(*Server).handleStream
        	            	  |   | 	external/org_golang_google_grpc/server.go:1712
        	            	  |   | google.golang.org/grpc.(*Server).serveStreams.func1.1
        	            	  |   | 	external/org_golang_google_grpc/server.go:947
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (13.91s)
=== RUN   TestDistSenderCircuitBreakerModes/scratchRange=false
    --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=false (33.25s)
=== RUN   TestDistSenderCircuitBreakerModes/scratchRange=false/mode=2
    test_server_shim.go:139: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    dist_sender_circuit_breaker_test.go:197: deadlocked range on n3 - r2:/System/NodeLiveness{-Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=4]
    dist_sender_circuit_breaker_test.go:230: attempt to scan node vitality resulted in unable to get liveness: aborted in DistSender: result is ambiguous: context deadline exceeded
    dist_sender_circuit_breaker_test.go:249: 
        	Error Trace:	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go:249
        	            				pkg/testutils/subtest.go:23
        	Error:      	Should be false
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=false/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=false/mode=2 (12.34s)
=== RUN   TestDistSenderCircuitBreakerModes/scratchRange=true
    --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true (41.71s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://tanzanite.cluster.engflow.com/invocations/default/c687761b-7cee-47eb-a5cf-bcb6a4153377?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on master @ [a23be6bce928c3e08074a15815b3c67a657bb40e](https://github.com/cockroachdb/cockroach/commits/a23be6bce928c3e08074a15815b3c67a657bb40e):


```
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
        	            	  |   | 	pkg/rpc/auth.go:90
        	            	  |   | google.golang.org/grpc.getChainUnaryHandler.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
        	            	  |   | 	pkg/rpc/context.go:132
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
        	            	  |   | 	pkg/util/stop/stopper.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
        	            	  |   | 	pkg/rpc/context.go:130
        	            	  |   | google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1145
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler
        	            	  |   | 	bazel-out/k8-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10679
        	            	  |   | google.golang.org/grpc.(*Server).processUnaryRPC
        	            	  |   | 	external/org_golang_google_grpc/server.go:1335
        	            	  |   | google.golang.org/grpc.(*Server).handleStream
        	            	  |   | 	external/org_golang_google_grpc/server.go:1712
        	            	  |   | google.golang.org/grpc.(*Server).serveStreams.func1.1
        	            	  |   | 	external/org_golang_google_grpc/server.go:947
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (14.94s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18058245?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18058245?buildTab=artifacts#/) on master @ [fa9acaa8a8b8b58b1448c61ca5a8f6e6a992d64c](https://github.com/cockroachdb/cockroach/commits/fa9acaa8a8b8b58b1448c61ca5a8f6e6a992d64c):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.RunValues[...].func1
        	            	  | 	pkg/testutils/subtest.go:23
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (2) aborted in DistSender
        	            	Wraps: (3) result is ambiguous: context deadline exceeded
        	            	Wraps: (4) secondary error attachment
        	            	  | aborted during DistSender.Send: context deadline exceeded
        	            	  | (1)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:3124
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2211
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).divideAndSendBatchToRanges
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1746
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1265
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*CrossRangeTxnWrapperSender).Send
        	            	  |   | 	pkg/kv/db.go:224
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender
        	            	  |   | 	pkg/kv/db.go:1133
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).send
        	            	  |   | 	pkg/kv/db.go:1116
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
        	            	  |   | 	pkg/kv/db.go:965
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Run
        	            	  |   | 	pkg/kv/db.go:988
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Get
        	            	  |   | 	pkg/kv/db.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestDistSenderCircuitBreakerModes.func1.1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go:205
        	            	  |   | github.com/cockroachdb/cockroach/pkg/testutils.RunValues[...].func1
        	            	  |   | 	pkg/testutils/subtest.go:23
        	            	  |   | testing.tRunner
        	            	  |   | 	GOROOT/src/testing/testing.go:1689
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (2) aborted during DistSender.Send
        	            	  | Wraps: (3) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (13.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18058557?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18058557?buildTab=artifacts#/) on master @ [21d1123dce5401b1ec580d4156db5db075074506](https://github.com/cockroachdb/cockroach/commits/21d1123dce5401b1ec580d4156db5db075074506):


```
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
        	            	  |   | 	pkg/rpc/auth.go:90
        	            	  |   | google.golang.org/grpc.getChainUnaryHandler.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1154
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
        	            	  |   | 	pkg/rpc/context.go:132
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
        	            	  |   | 	pkg/util/stop/stopper.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
        	            	  |   | 	pkg/rpc/context.go:130
        	            	  |   | google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1
        	            	  |   | 	external/org_golang_google_grpc/server.go:1145
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler
        	            	  |   | 	bazel-out/aarch64-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10679
        	            	  |   | google.golang.org/grpc.(*Server).processUnaryRPC
        	            	  |   | 	external/org_golang_google_grpc/server.go:1335
        	            	  |   | google.golang.org/grpc.(*Server).handleStream
        	            	  |   | 	external/org_golang_google_grpc/server.go:1712
        	            	  |   | google.golang.org/grpc.(*Server).serveStreams.func1.1
        	            	  |   | 	external/org_golang_google_grpc/server.go:947
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_arm64.s:1222
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (12.68s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvclient/kvcoord.TestDistSenderCircuitBreakerModes [failed](https://tanzanite.cluster.engflow.com/invocations/default/2e535912-ee74-45a5-8f9b-751f13686c5c?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on master @ [21d1123dce5401b1ec580d4156db5db075074506](https://github.com/cockroachdb/cockroach/commits/21d1123dce5401b1ec580d4156db5db075074506):


```
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1746
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1265
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*CrossRangeTxnWrapperSender).Send
        	            	  |   | 	pkg/kv/db.go:224
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender
        	            	  |   | 	pkg/kv/db.go:1133
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).send
        	            	  |   | 	pkg/kv/db.go:1116
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
        	            	  |   | 	pkg/kv/db.go:965
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Run
        	            	  |   | 	pkg/kv/db.go:988
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Get
        	            	  |   | 	pkg/kv/db.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestDistSenderCircuitBreakerModes.func1.1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go:205
        	            	  |   | github.com/cockroachdb/cockroach/pkg/testutils.RunValues[...].func1
        	            	  |   | 	pkg/testutils/subtest.go:23
        	            	  |   | testing.tRunner
        	            	  |   | 	GOROOT/src/testing/testing.go:1689
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (5) r74/3:(n3,s3) is unavailable (circuit breaker tripped)
        	            	  | Wraps: (6) forced error mark
        	            	  |   | "originated at breaker r74/3:(n3,s3)"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (7) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (8)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).sendProbe
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:1002
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*ReplicaCircuitBreaker).launchProbe.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker.go:875
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  |   | 	pkg/util/stop/stopper.go:498
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (9) probe timed out
        	            	  | Wraps: (10) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errbase.opaqueWrapper (3) *errutil.withPrefix (4) *errbase.opaqueWrapper (5) *errutil.withPrefix (6) *markers.withMark (7) *markers.withMark (8) *errbase.opaqueWrapper (9) *errutil.withPrefix (10) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2
        --- FAIL: TestDistSenderCircuitBreakerModes/scratchRange=true/mode=2 (18.78s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderCircuitBreakerModes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go
@@ -138,10 +138,14 @@ func TestDistSenderCircuitBreakerModes(t *testing.T) {
 			func(t *testing.T, mode kvcoord.DistSenderCircuitBreakersMode) {
 				ctx := context.Background()
 
-				// The lease won't move unless we use expiration-based leases. We also
-				// speed up the test by reducing various intervals and timeouts.
+				// The lease won't move unless we use expiration-based leases. This is
+				// because a deadlocked range with expiration-based leases would
+				// eventually cause the lease to expire. However, with epoch-based or
+				// leader leases, the lease wouldn't expire due to a deadlocked range
+				// since lease extensions are happening at a different layer.
+				// We also speed up the test by reducing various intervals and timeouts.
 				st := cluster.MakeTestingClusterSettings()
-				kvserver.ExpirationLeasesOnly.Override(ctx, &st.SV, true)
+				kvserver.OverrideDefaultLeaseType(ctx, &st.SV, roachpb.LeaseExpiration)
 				kvcoord.CircuitBreakersMode.Override(ctx, &st.SV, mode)
 				kvcoord.CircuitBreakerCancellation.Override(ctx, &st.SV, true)
 				kvcoord.CircuitBreakerProbeThreshold.Override(ctx, &st.SV, time.Second)

